### ISSUE TITLE
Qhub local deployment failed [bug]

### ISSUE THREAD

**Issue by viniciusdc** [Quansight/qhub-cloud#443]
I've created a clean clone of qhub-cloud to test the subdir issue fix. Before I changed anything within the Qhub code, I tried to deploy it as usual (following the [docs](https://github.com/Quansight/qhub-cloud/blob/dev/docs/source/06_developers_contrib_guide/04_tests.md) ) but when executing the last step 
`python -m qhub deploy --config qhub-config.yaml --disable-prompt
`
I received the following error:

```python
INFO:qhub.deploy:All qhub endpoints will be under https://github-actions.qhub.dev
INFO:qhub.provider.terraform:terraform=/tmp/terraform/0.13.5/terraform init directory=infrastructure
INFO:qhub.provider.terraform:terraform init took 0.612 [s]
INFO:qhub.provider.terraform:terraform=/tmp/terraform/0.13.5/terraform apply directory=infrastructure targets=['module.kubernetes', 'module.kubernetes-initialization', 'module.kubernetes-ingress']

Error: namespaces "dev" already exists


ERROR:qhub.deploy:b'\x1b[0m\x1b[1mmodule.kubernetes-initialization.kubernetes_namespace.main: Creating...\x1b[0m\x1b[0m\n\x1b[33m\n\x1b[1m\x1b[33mWarning: \x1b[0m\x1b[0m\x1b[1mResource targeting is in effect\x1b[0m\n\n\x1b[0mYou are creating a plan with the -target option, which means that the result\nof this plan may not represent all of the changes requested by the current\nconfiguration.\n\t\t\nThe -target option is not for routine use, and is provided only for\nexceptional situations such as recovering from errors or mistakes, or when\nTerraform specifically suggests to use it as part of an error message.\n\x1b[0m\x1b[0m\n\x1b[33m\n\x1b[1m\x1b[33mWarning: \x1b[0m\x1b[0m\x1b[1mApplied changes may be incomplete\x1b[0m\n\n\x1b[0mThe plan was created with the -target option in effect, so some changes\nrequested in the configuration may have been ignored and the output values may\nnot be fully updated. Run the following command to verify that no other\nchanges are pending:\n    terraform plan\n\t\nNote that the -target option is not suitable for routine use, and is provided\nonly for exceptional situations such as recovering from errors or mistakes, or\nwhen Terraform specifically suggests to use it as part of an error message.\n\x1b[0m\x1b[0m\n\x1b[33m\n\x1b[1m\x1b[33mWarning: \x1b[0m\x1b[0m\x1b[1mInterpolation-only expressions are deprecated\x1b[0m\n\n\x1b[0m  on .terraform/modules/qhub/modules/kubernetes/services/meta/qhub/main.tf line 10, in module "kubernetes-jupyterhub":\n  10:           \x1b[4m"${var.general-node-group.key}"\x1b[0m = var.general-node-group.value\n\x1b[0m\nTerraform 0.11 and earlier required all non-constant expressions to be\nprovided via interpolation syntax, but this pattern is now deprecated. To\nsilence this warning, remove the "${ sequence from the start and the }"\nsequence from the end of this expression, leaving just the inner expression.\n\nTemplate interpolation syntax is still used to construct strings from\nexpressions when the template includes multiple interpolation sequences or a\nmixture of literal strings and interpolations. This deprecation applies only\nto templates that consist entirely of a single interpolation sequence.\n\n(and 3 more similar warnings elsewhere)\n\x1b[0m\x1b[0m\n'
Traceback (most recent call last):
  File "/home/vinicius/miniconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/vinicius/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/__main__.py", line 11, in <module>
    main()
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/__main__.py", line 7, in main
    cli(sys.argv[1:])
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/cli/__init__.py", line 40, in cli
    args.func(args)
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/cli/deploy.py", line 59, in handle_deploy
    config, args.dns_provider, args.dns_auto_provision, args.disable_prompt
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/deploy.py", line 24, in deploy_configuration
    raise e
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/deploy.py", line 21, in deploy_configuration
    guided_install(config, dns_provider, dns_auto_provision, disable_prompt)
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/deploy.py", line 54, in guided_install
    "module.kubernetes-ingress",
  File "/home/vinicius/Quansight/local/qhub-cloud/qhub/provider/terraform.py", line 82, in apply
    cwd=directory,
  File "/home/vinicius/miniconda3/lib/python3.7/subprocess.py", line 411, in check_output
    **kwargs).stdout
  File "/home/vinicius/miniconda3/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/tmp/terraform/0.13.5/terraform', 'apply', '-auto-approve', '-target=module.kubernetes', '-target=module.kubernetes-initialization', '-target=module.kubernetes-ingress']' returned non-zero exit status 1.
```
`Terraform v0.13.5`
`minikube version: v1.17.1`

**Comment by viniciusdc** [comment]
I didn't use the render code (as it's merged with deploy for standard test). I also tested with the same minikube server with an outdated version of Qhub (another repo with updates from one week ago) and it worked...

**Comment by leej3** [comment]
I also encountered this error. It can be side-stepped by adding --namespace to the `qhub init` command

minikube version: v1.18.1
Terraform (downloaded during deployment is also 0.13.5)
Qhub: a4a479abf7e099c202c0c9a426197b7d66c76f9c on dev branch

**Comment by viniciusdc** [comment]
> I also encountered this error. It can be side-stepped by adding --namespace to the `qhub init` command
> 
> minikube version: v1.18.1
> Terraform (downloaded during deployment is also 0.13.5)
> Qhub: [a4a479a](https://github.com/Quansight/qhub-cloud/commit/a4a479abf7e099c202c0c9a426197b7d66c76f9c) on dev branch

But then you found the error ` terraform-traefik-traefik" already exists` right?

**Comment by viniciusdc** [comment]
@costrouc

**Comment by leej3** [comment]
>  you found the error terraform-traefik-traefik" already exists right

Indeed. I don't have enough knowledge of the system to commit to that though :) 

This is was occurring along with an issue where the downloaded terraform binary was not used. This was  resolved by @costrouc's fix in 1d804b762fa69f18f4528f225ba104a1a78226cd

The issue occued because of a pre-existing cluster. It could be resolved by using `qhub destroy` with the appropriate config file.

**Comment by viniciusdc** [comment]
solved
